### DISCUSSION TITLE
45. VATZ biweekly meeting at `2023-10-25 08:00:00 -05:00:00`

### DISCUSSION THREAD

**Discussion by xellos00** [dsrvlabs/vatz#510] [General]
### 1. Overall

<img width="1213" alt="image" src="https://github.com/dsrvlabs/vatz/assets/6308023/d3d7dc40-657b-4d91-90a2-84b628643966">

### 2. Statistic Rate
 
Sprint | Issue fulfillment | progress rate(%)
--: | :--: | :--:
43 | 2/14 | 14%
44 | 2/14 | 14%

---

<img width="1248" alt="image" src="https://github.com/dsrvlabs/vatz/assets/6308023/43268365-26a0-4446-b6d5-0b58df6e7456">

**Comment by xellos00** [comment]
# DK

- https://github.com/dsrvlabs/vatz/issues/498
- https://github.com/dsrvlabs/vatz/issues/506
- https://github.com/dsrvlabs/vatz/issues/509
- https://github.com/dsrvlabs/vatz/issues/511
- https://github.com/dsrvlabs/vatz/issues/515
- https://github.com/dsrvlabs/vatz/issues/520
- https://github.com/dsrvlabs/vatz/issues/521
- https://github.com/dsrvlabs/vatz/issues/519

**Comment by rootwarp** [comment]
# 준교

## Testing gRPC Reflection

WIP Architecture below.

![image](https://github.com/dsrvlabs/vatz/assets/9974087/7d73bcf1-ac06-4048-b80b-73df63d97005)

Implement two components for testing.
1. VATZ Agent
2. VATZ Plugin

Scenario
1. Agent: Start and wait to receive registration request
  - [x] Open localhost:8080 to receive registration request, and it should be gRPC service.
  - [x] Registration service provides `RegistPlugin` service with `name`,  `address` and `port`.
2. Plugin: Start and call `RegisterPlugin`.
  - [x] it will start gRPC service to provide plugin feature. and will be localhost:9090 
  - [x] it provides `HelloThere` function.
3. Agent
  - [x] Connect to plugin service and request reflections.
  - [x] Marshaling reflect specs and test re-structuring
  - [ ] Handle and manage "spec" data.
  - [ ] Call plugin function -> Partially completed.
---
Next
- Research how do we setup a protocol between user and VATZ service.
  - Need to provide plugin informations, list services, query call specs, etc...
  - Define a message format to call a plugin.

**Comment by kim201212** [comment]
# 형기

- https://github.com/dsrvlabs/vatz/pull/516

**Comment by skonhwang** [comment]
## 승곤
- no issues

**Comment by gnongs** [comment]
## 진홍
- no issue
